### PR TITLE
test: Cache most commonly used VC contexts for tests

### LINKED ISSUE #1753: Cache most commonly used VC contexts for tests
Currently, the base `https://www.w3.org/2018/credentials/v1` is cached by default, we should use CachingJSONLDLoader() and add caches for most commonly used contexts (like `https://www.w3.org/2018/credentials/examples/v1` and `https://trustbloc.github.io/context/vc/examples-v1.jsonld`). This will speed-up the execution of the tests and prevent sporadical test breaks (e.g. in case of network unavailability).

See https://github.com/hyperledger/aries-framework-go/pull/1743#discussion_r417853752.

### PR DESCRIPTION
**Pull request by kdimak**
closes #1753

Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

In a follow-up PR, the same will be done for linked data proofs (`signature` package).

### CODE REVIEW

File: pkg/doc/verifiable/credential_ldp_test.go [hunk 1]
```diff
@@ -204,13 +204,41 @@ func TestNewCredentialFromLinkedDataProof_JSONLD_Validation(t *testing.T) {
...
 
-		vcWithLdp, _, err := NewCredential([]byte(vcJSON), vcOptions...)
+		vcWithLdp, _, err := newTestCredential([]byte(vcJSON), vcOptions...)
 		r.Error(err)
 		r.EqualError(err, "JSON-LD doc has different structure after compaction")
 		r.Nil(vcWithLdp)
 	})
 
 	t.Run("VC with different mapped field", func(t *testing.T) {
+		mavennetJSONLD := `
```
Comment (troyronda) [inline pkg/doc/verifiable/credential_ldp_test.go:214]: should replace with our own simple version that tests the changes property name scenario.

Comment (troyronda) [reply pkg/doc/verifiable/credential_ldp_test.go:214]: e.g., a simple context that includes "comments": "schema:text"

File: pkg/doc/verifiable/support_test.go [hunk 2]
```diff
@@ -354,3 +356,106 @@ func createVCWithTwoLinkedDataProofs() (*Credential, PublicKeyFetcher) {
...
+//nolint:gochecknoglobals
+var testDocumentLoader = createTestJSONLDDocumentLoader()
+
+func createTestJSONLDDocumentLoader() *ld.CachingDocumentLoader {
+	fmt.Printf("Create test document loader!\n")
+
+	loader := CachingJSONLDLoader()
+
+	const exampleJSONLDContext = `
+{
```
Comment (troyronda) [inline pkg/doc/verifiable/support_test.go:369]: it would be nice to move these test contexts into files under the `testdata` directory.

File: pkg/doc/verifiable/credential_ldp_test.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/verifiable/credential_ldp_test.go]: done!

File: pkg/doc/verifiable/support_test.go (hunk 2 above)
Comment (kdimak) [reply pkg/doc/verifiable/support_test.go]: done!

File: pkg/doc/verifiable/credential_ldp_test.go [hunk 3]
```diff
@@ -595,6 +590,29 @@ func TestNewCredentialFromLinkedDataProof_EcdsaSecp256k1Signature2019(t *testing
 
 //nolint:lll
 func TestNewCredential_JSONLiteralsNotSupported(t *testing.T) {
+	cmtrJSONLD := `
+{
+  "@context": {
```
Comment (troyronda) [inline pkg/doc/verifiable/credential_ldp_test.go:595]: can this be moved to testdata file?

File: pkg/doc/verifiable/credential_ldp_test.go [hunk 4]
```diff
@@ -595,6 +590,29 @@ func TestNewCredentialFromLinkedDataProof_EcdsaSecp256k1Signature2019(t *testing
 
 //nolint:lll
 func TestNewCredential_JSONLiteralsNotSupported(t *testing.T) {
+	cmtrJSONLD := `
```
Comment (troyronda) [inline pkg/doc/verifiable/credential_ldp_test.go:593]: the below isn't a `cmtr`.

File: pkg/doc/verifiable/credential_ldp_test.go [hunk 5]
```diff
@@ -595,6 +590,29 @@ func TestNewCredentialFromLinkedDataProof_EcdsaSecp256k1Signature2019(t *testing
...
+    "cmtr": {
+      "@id": "hetc:cmtr",
+      "@type": "@json"
+    }
+  }
+}
+`
+
+	docLoader := createTestJSONLDDocumentLoader()
+	addJSONLDCachedContext(docLoader, "https://www.w3.org/2018/credentials/examples/v1", cmtrJSONLD)
```
Comment (troyronda) [inline pkg/doc/verifiable/credential_ldp_test.go:614]: why `https://www.w3.org/2018/credentials/examples/v1` - doesn't seem related to the context.

File: pkg/doc/verifiable/credential_ldp_test.go [hunk 6]
```diff
@@ -595,6 +590,29 @@ func TestNewCredentialFromLinkedDataProof_EcdsaSecp256k1Signature2019(t *testing
...
+  "@context": {
+    "@version": 1.1,
+    "@protected": true,
+    "name": "http://schema.org/name",
+    "description": "http://schema.org/description",
+    "image": {
+      "@id": "http://schema.org/image",
+      "@type": "@id"
+    },
+    "hetc": "https://w3c-ccg.github.io/vc-json-schemas/example/cmtr/v0.2/cmtr-v0.2#",
```
Comment (troyronda) [inline pkg/doc/verifiable/credential_ldp_test.go:604]: can we remove these references to w3c-ccg. it doesn't seem needed for the test.

Comment (kdimak) [reply pkg/doc/verifiable/credential_ldp_test.go:604]: right, I will simplify that!

File: pkg/doc/verifiable/credential_ldp_test.go [hunk 7]
```diff
@@ -204,46 +204,40 @@ func TestNewCredentialFromLinkedDataProof_JSONLD_Validation(t *testing.T) {
...
+		vcWithLdp, _, err := newTestCredential([]byte(vcJSON), vcOptions...)
 		r.Error(err)
 		r.EqualError(err, "JSON-LD doc has different structure after compaction")
 		r.Nil(vcWithLdp)
 	})
 
 	t.Run("VC with different mapped field", func(t *testing.T) {
+		localJSONLDContext := `
+{
+  "@context":
```
Comment (troyronda) [inline pkg/doc/verifiable/credential_ldp_test.go:216]: I assume this is still inline due to being tiny?

Comment (kdimak) [reply pkg/doc/verifiable/credential_ldp_test.go:216]: Yeah, this is tiny enough file to leave it inlined.

File: pkg/doc/verifiable/credential_ldp_test.go (hunk 5 above)
Comment (kdimak) [reply pkg/doc/verifiable/credential_ldp_test.go]: It was a bug - corrected.

File: pkg/doc/verifiable/credential_ldp_test.go (hunk 6 above)
Comment (kdimak) [reply pkg/doc/verifiable/credential_ldp_test.go]: Simplified - used `http://127.0.0.1:53401/cmtr.jsonld` dummy context, generated a new valid signature due to `"hetc": "http://localhost:9393/cmtr#"` simplification (no reference to `w3c-ccg`).
